### PR TITLE
LP-11431.6: Clean up KAS package

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
This PR makes the following changes to KAS:
- removes unnecessary dev deps
- updates yarn.lock
- removes duplicated files that were moved (parser.js, unitparser.js - these are in `__genfiles__` now)
- removes unnecessary files (test.html, lint_blacklist.txt)

Issue: LP-11431

## Test plan:
- let GitHub Actions run all the checks

### CODE REVIEW

**Comment by kevinbarabash** [comment]
```
expect("f(x) = sin^2(x)+cos^2(x)").toEqualExpr("f(x) = 1");
```
appears to be a flaky test.  I saw some flakiness when using wallaby, but I thought it was due to wallaby and not the test itself.    I'll dig in and try to determine the cause.

File: .eslintrc.js [hunk 1]
```diff
@@ -5,9 +5,6 @@ const path = require("path");
 const pkgAliases = fs
     .readdirSync(path.join(__dirname, "packages"))
     .map((pkgName) => {
-        if (pkgName === "kas") {
```
Comment (jeremywiebe) [inline .eslintrc.js:8]: Great. I like that we are keeping our configs largely the same for all packages!

File: packages/kas/package.json [hunk 2]
```diff
@@ -29,8 +29,6 @@
     "underscore": "1.4.4"
```
Comment (jeremywiebe) [inline packages/kas/package.json:29]: Should this be a peer/dev dependency?

File: packages/kas/src/__tests__/comparing_test.js [hunk 3]
```diff
@@ -259,7 +259,20 @@ describe("comparing", () => {
...
+        // between the expressions is always less than some very small number.
+        // In this case though, we convert equations to expressions, e.g.
+        // `f(x) = 1` is converted to `1-f(x)`, but because `f` is a function,
+        // we can't evaluate it.  One possible solution to this would be
+        // to isolate `f(x)` in each expression and then check that the
+        // expressions that they're equal to are equal.  In this case that
+        // would be `sin^2(x)+cos^2(x)` and `1`.
+        // TODO(TP-11651): Update compare to isolate functions on wide side
+        // before comparing expressions on the other side.
+        // expect("f(x) = sin^2(x)+cos^2(x)").toEqualExpr("f(x) = 1");
```
Comment (jeremywiebe) [inline packages/kas/src/__tests__/comparing_test.js:275]: Thanks for leaving this detailed comment!

File: packages/kas/package.json (hunk 2 above)
Comment (kevinbarabash) [reply packages/kas/package.json:29]: Good call.

**Comment by kevinbarabash** [comment]
The parent pull-request ([#184](https://github.com/Khan/perseus/pull/184)) has been merged into `main`, but this branch (`lp-11431.6`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

**Pull request merged**